src/factory.rs

```diff
@@ -9,16 +9,16 @@ pub fn factory<C: RuntimeName, I: ComInterface>() -> Result<I> {
 
     unsafe {
         // First attempt to get the activation factory via the OS.
-        let mut code = RoGetActivationFactory(name.get_abi(), &I::iid(), &mut factory);
+        let mut code = RoGetActivationFactory(name.get_abi(), &I::iid(), &mut factory)?;
```

Comment:
Short circuiting here won't allow reg-free activation of WinRT components on Windows 7. https://github.com/microsoft/winrt-rs/pull/177

Reply:
Huh, didn't even consider that you could do that!
Looks like it would be fine to just squish the delay-load error into the regular one?

Reply:
Yes, if it fails to load or fails to activate we can treat it the same regardless.